nipype/algorithms/confounds.py

```diff
@@ -1067,7 +1067,7 @@ def compute_noise_components(imgseries, mask_images, num_components,
 
         # "The covariance matrix C = MMT was constructed and decomposed into its
         # principal components using a singular value decomposition."
-        u, _, _ = linalg.svd(M, full_matrices=False)
+        u, _, _ = linalg.svd(M[~np.isnan(M)], full_matrices=False)
```

Comment:
i don't see how anything in `M` would be a `nan` given line 1047 and the `_compute_tSTD` function which should return a 1 if the variance is 0.

is there a particular use case we can look at where you found this?

Reply:
I'm going to look closer into this. You're right. And this change would reshape `M` so it basically breaks the interface.